HIVE-26658: INT64 Parquet timestamps cannot be mapped to most Hive numeric types

zabetak opened the pull request:
### What changes were proposed in this pull request?
1. Unify converters from Parquet INT64 to Hive types.
2. Add tests reading from Parquet INT64 timestamp to various Hive numeric types.

### Why are the changes needed?
Restore backward compatibility; allow mapping INT64 timestamps with timestamp annotation to the following Hive numeric types:
* TINYINT
* SMALLINT
* INT
* DOUBLE
* FLOAT
* DECIMAL

For more details see HIVE-26658.

### Does this PR introduce _any_ user-facing change?
Avoids errors/exceptions when attempting to map Parquet INT64 with timestamp to anything except TIMESTAMP & BIGINT.

### How was this patch tested?
```
mvn test -Dtest=TestETypeConverter
mvn test -Dtest=TestMiniLlapLocalCliDriver -Dqfile=parquet_int64_timestamp_to_numeric.q
```

zabetak commented:
Still relevant, please keep this open!

cnauroth reviewed (commented):
Hello @zabetak . This looks good to me, though the CI run from a few months ago showed a test failure. Is it because the .q and .q.out files have a different ordering for the `ALTER TABLE` statements?

zabetak commented:
Thanks for the review @cnauroth ! Indeed I changed the order of the statements in the .q file just before opening the PR and forgot to update the respective .q.out.

I rebased the PR against latest master and updated the stale .q.out file so I am hoping that now all tests will come back green.

zabetak commented:
Hey @scarlin-cloudera , since this is a follow-up of a ticket that you worked on can you please have a look as well?

cnauroth reviewed (approved):
The tests passed on the last run, so +1 (non-binding). Thanks for the update, @zabetak !

scarlin-cloudera commented:
LGTM

ayushtkn reviewed (approved):
LGTM
